smartapps/encored-technologies/smart-energy-service.src/smart-energy-service.groovy

```diff
@@ -0,0 +1,1388 @@
+/**
+ *  ProtoType Smart Energy Service
+ *
+ *  Copyright 2015 hyeon seok yang
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+ 
+definition(
+    name: "Smart Energy Service",
+    namespace: "Encored Technologies",
+    author: "hyeon seok yang",
+    description: "With visible realtime energy usage status, have good energy habits and enrich your life\r\n",
+    category: "SmartThings Labs",
+    iconUrl: "https://s3-ap-northeast-1.amazonaws.com/smartthings-images/appicon_enertalk%401.png",
+    iconX2Url: "https://s3-ap-northeast-1.amazonaws.com/smartthings-images/appicon_enertalk%402x",
+    iconX3Url: "https://s3-ap-northeast-1.amazonaws.com/smartthings-images/appicon_enertalk%403x",
+    oauth: true)
+{
+    appSetting "clientId"
+    appSetting "clientSecret"
+    appSetting "callback"
+}
+
+
+preferences {
+	page(name: "checkAccessToken")
+}
+
+cards {
+    card(name: "Encored Energy Service", type: "html", action: "getHtml", whitelist: whiteList()) {}
+}
+
+/* This list contains, that url need to be allowed in Smart Energy Service.*/
+def whiteList() {
+	[
+    	"code.jquery.com", 
+    	"ajax.googleapis.com", 
+        "fonts.googleapis.com",
+    	"code.highcharts.com", 
```

Comment:
"code.highcharts.com" - can this JS be stored on a more reliable domain such as AWS? Is this even necessary?